### PR TITLE
[FIX] website_sale: prevent zero price display if attributes

### PR DESCRIPTION
**Pull request by thle-odoo**
Steps to reproduce:
-------------------
- activate "Prevent Sale of Zero Priced Product" setting;
- create a product with a price equals to zero;
- add attribute for this product;
- go to ecommerce.

Issue:
------
The sentence "Not Available For Sale" is displayed on the shop page and in the search bar.
This is not correct as the product is available.
It has a price thanks to these attributes.

Solution:
---------
Use the `attribute_line_ids` to detect
if the product has attributes or not.

opw-3374137

### CODE REVIEW

**Comment by thle-odoo** [comment]
Hello @vava-odoo 

Would it make sense to test if the "Value price Extra" of the attribute is non-zero ?
By this I mean not only checking that the product has attributes, but also that there is at least one attribute with a non-zero price to not display the "Not Available For Sale" message.
This would make the fix more invasive, but I prefer to ask :)

**Comment by vava-odoo** [comment]
Not sure we want to "correct" this flow... Asking @paju-odoo for a functional decision here or on the ticket

**Comment by paju-odoo** [comment]
Hi @thle-odoo,

I wouldn't change the current behavior....

Businesses who use this setting mainly target B2B customers and want to display a catalog, without any prices. They use it to avoid displaying any price in general, they don't want to display it for specific variants. 

Furthermore, I do think we provide efficient features but this is also the user's responsibility to make wise choices. Why would you have "free" products that are no longer free for certain variants? And in that case, you shouldn't use this setting.

I don't even see what price could be displayed in /shop and search... The first variant whose price is not equal to 0? That wouldn't make sense, since no variant is selected by the user at that moment. Clicking on the product template in /shop redirects the user to the first variant, which then has a price equal to zero... This would be very confusing for the customer.

In summary, I think this is a too complicated and niche flow to make it a generic behavior. I can't think of any relevant use case...

cc @vava-odoo

**Comment by thle-odoo** [comment]
Hello @paju-odoo,

Thank you for these explanations (I didn't have this point of view).
I understand that this flow must not be modified.

I therefore close this PR.
